analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UriShouldNotBeHardcoded.CSharp11.cs

```diff
@@ -17,22 +17,29 @@ public void RawStrings()
         var unixPath2 = """~/blah/blah/blah.txt"""; // Compliant
         var unixPath3 = """~\\blah\\blah\\blah.txt"""; // Compliant
         var windowsPathStartingWithVariable = """%AppData%\\Adobe""";
-
-        var webChemin = """http://www.mywebsite.com"""; // FN
-        var windowsChemin = """c:\\blah\\blah\\blah.txt"""; // FN
     }
 
     public void SpanMatch(Span<char> span, ReadOnlySpan<char> readonlySpan, string simpleString)
     {
-        var a = span is """\\my-network-drive\folder\file.txt"""               // FN
-                || readonlySpan is """\\my-network-drive\folder\file.txt"""    // FN
-                || simpleString is @"\\my-network-drive/folder/file.txt";      // FN
+        var a = span is """\\my-network-drive\folder\file.txt"""               // Compliant (variable name 'a' not containing any checked strings)
+                || readonlySpan is """\\my-network-drive\folder\file.txt"""    // Compliant (variable name 'a' not containing any checked strings)
+                || simpleString is @"\\my-network-drive/folder/file.txt";      // Compliant (variable name 'a' not containing any checked strings)
 
         var URI = span is """\\my-network-drive\folder\file.txt"""             // Noncompliant
                 || readonlySpan is """\\my-network-drive\folder\file.txt"""    // Noncompliant
                 || simpleString is @"\\my-network-drive/folder/file.txt";      // Noncompliant
     }
 
-    public bool ListPattern(string[] uris) =>
-        uris is ["""\\my-network-drive\folder\file.txt"""]; // FN
+    public void ListPattern(string[] uris)
+    {
+        bool pathFlag = uris is ["""\\my-network-drive\folder\file.txt"""]; // Noncompliant
+    }
+
+    public bool ReturnStement(string uri)
+    {
+        return uri is """\\my-network-drive\folder\file.txt"""; // Compliant (not raising for return statement)
```

Comment:
Please mark this and the next example as FN. 
Please also add a comment that this rule is not raising here because it only checks the string literals that are
1 - Arguments in methods `Request.MapPath(@"C:\path\stuff.txt"); // Noncompliant (n.b. method will throw exception)`
2 - Arguments in constructors ` new Microsoft.AspNetCore.Mvc.VirtualFileResult(@"C:\path\stuff.txt", "text/javascript"); // Noncompliant`
3 - values assigned to variables `var webUri1 = "http://www.mywebsite.com"; // Noncompliant`

My assumption is that we don't check every single literal or generally other cases so we don't create too much noise (too many FPs.)

Reply:
I've marked the test cases as FNs and moved them into the base test class since they are not C#11 related.